src/SourceCode.Chasm.Repository.AzureBlob/AzureBlobChasmRepo.CommitRef.cs

```diff
@@ -63,49 +63,50 @@ private async ValueTask<(CommitRef, AccessCondition)> ReadCommitRefImplAsync(Clo
             var blobName = DeriveCommitRefBlobName(branch, name);
             var blobRef = refsContainer.GetAppendBlobReference(blobName);
 
-            var (commitRef, _) = await ReadCommitRefImplAsync(blobRef, cancellationToken).ConfigureAwait(false);
+            var (commitId, _) = await ReadCommitRefImplAsync(blobRef, cancellationToken).ConfigureAwait(false);
+
+            var commitRef = new CommitRef(name, commitId);
             return commitRef;
         }
 
         #endregion
 
         #region Write
 
-        public async Task WriteCommitRefAsync(CommitId? previousCommitId, string branch, string name, CommitRef commitRef, CancellationToken cancellationToken)
+        public async Task WriteCommitRefAsync(CommitId? previousCommitId, string branch, CommitRef commitRef, CancellationToken cancellationToken)
         {
             if (string.IsNullOrWhiteSpace(branch)) throw new ArgumentNullException(nameof(branch));
-            if (string.IsNullOrWhiteSpace(name)) throw new ArgumentNullException(nameof(name));
+            if (commitRef == CommitRef.Empty) throw new ArgumentNullException(nameof(commitRef));
 
             var refsContainer = _refsContainer.Value;
 
-            var blobName = DeriveCommitRefBlobName(branch, name);
+            var blobName = DeriveCommitRefBlobName(branch, commitRef.Name);
             var blobRef = refsContainer.GetAppendBlobReference(blobName);
 
-            // Load existing commit ref in order to use its ETAG
-            var (existingCommitRef, accessCondition) = await ReadCommitRefImplAsync(blobRef, cancellationToken).ConfigureAwait(false);
+            // Load existing commit ref in order to use its etag
+            var (existingCommitId, ifMatchCondition) = await ReadCommitRefImplAsync(blobRef, cancellationToken).ConfigureAwait(false);
 
-            if (existingCommitRef != CommitRef.Empty)
+            if (existingCommitId != CommitId.Empty)
```

Comment:
Remove this condition. You must use an empty commit ID if the previous commit ID is empty.

Reply:
No that check tells us whether or not the CommitRef was found in storage at all. But I agree it's ambiguous, will change the pattern